### PR TITLE
S3 notification event trigger Java-based AWS Lambda function

### PR DESCRIPTION
**Pull request by dmahapatro**
P144345160

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by yualexan** [comment]
Thanks @dmahapatro for the quick turnaround.

@pputhran @julianwood Can you please help review or identify reviewers for this?

**Comment by seshubaws** [comment]
Thank you @dmahapatro for adding this! Could you add an example-pattern.json file to this? Here's a [sample example-pattern.json](https://github.com/aws-samples/serverless-patterns/blob/main/_pattern-model/example-pattern.json).

**Comment by dmahapatro** [comment]
@seshubaws Added `example-pattern.json`.

File: s3-lambda-resizing-java/template.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: Image resizing service with Java-based Lambda functions
+
+Parameters:
+  SourceBucketName:
+    Type: String
```
Comment (seshubaws) [inline s3-lambda-resizing-java/template.yaml:7]:
```suggestion
    Type: String
    # Pass your bucket name dynamically or change it here
    Default: my-source-bucket-name
```

File: s3-lambda-resizing-java/template.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  DestinationBucketName:
+    Type: String
```
Comment (seshubaws) [inline s3-lambda-resizing-java/template.yaml:9]:
```suggestion
    Type: String
    # Pass your bucket name dynamically or change it here
    Default: my-destination-bucket-name
```

File: s3-lambda-resizing-java/README.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    cd s3-lambda-resizing-java
+    ```
+- From the command line, use AWS SAM to build and deploy the AWS resources for the pattern as specified in the `template.yml` file:
+    ```
+    sam build && sam deploy --guided
+    ```
+- During the prompts:
+  * Enter a stack name
+  * Enter the desired AWS Region
+  * Allow SAM CLI to create IAM roles with the required permissions.
```
Comment (seshubaws) [inline s3-lambda-resizing-java/README.md:44]:
```suggestion
  * Allow SAM CLI to create IAM roles with the required permissions.
      * Enter names for your source and destination S3 buckets. Make sure these are unique as S3 bucket names share a global namespace.
```

File: s3-lambda-resizing-java/README.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
+# AWS Amazon S3 to AWS Lambda — Create a Java-based Lambda function that resizes images uploaded to S3
```
Comment (ellisms) [inline s3-lambda-resizing-java/README.md:1]:
```suggestion
# Resizing images uploaded to Amazon S3 with AWS Lambda in Java
```

File: s3-lambda-resizing-java/README.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+Run the following command to check that a new version of the image has been created in the destination bucket.
+
+```bash
+aws s3 ls s3://{DestinationBucketName}
+```
+
+## Running JUnit 5 test
+To run the JUnit test `AppTest.java`, you need to make few changes:
+
+- Modify the [events/s3_event.json](./ResizerFunction/src/test/resources) file to reflect the source bucket name and object key. Replace these values in the placeholders.
```
Comment (ellisms) [inline s3-lambda-resizing-java/README.md:76]: Can we default the object key to white_dog.jepg?

File: s3-lambda-resizing-java/README.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+- Replace the destination bucket name placeholder in the surefire plugin's environment variable placeholder in the `pom.xml` file.
+- Enable the test [`AppTest.java`](./ResizerFunction/src/test/java/resizer/AppTest.java) by removing the `@Disabled` annotation.
+- Run `mvn clean test`
```
Comment (ellisms) [inline s3-lambda-resizing-java/README.md:79]: Add changing directory to ./ResizerFunction

File: s3-lambda-resizing-java/README.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+The Lambda code checks the uploaded file is an image and creates a lower resolution version of the image in the destination bucket.
+
+Learn more about this pattern at Serverless Land Patterns: [https://serverlessland.com/patterns/s3-lambda](https://serverlessland.com/patterns/s3-lambda)
+
+Important: this application uses various AWS services, and there are costs associated with these services after the Free Tier usage.
+Please see the [AWS Pricing page](https://aws.amazon.com/pricing/) for details. 
+You are responsible for any AWS costs incurred. No warranty is implied in this example.
+
+## Requirements
+
```
Comment (ellisms) [inline s3-lambda-resizing-java/README.md:21]: Add requirement for Java21

**Review by ellisms — CHANGES_REQUESTED** [review]

**Comment by dmahapatro** [comment]
@ellisms Review comments are incorporated.

**Comment by ellisms** [comment]
Thanks @dmahapatro. Adding this to the publishing queue.

**Review by ellisms — APPROVED** [review]

**Pull request merged**